OpenTelemetry (S3M) Host and OS apps (#1878)
SumoLogic/sumologic-documentation: docs/integrations/hosts-operating-systems/opentelemetry/windows-opentelemetry.md

sidebar_label: Windows - OTel Collector
description: Learn about the Sumo Logic OpenTelemetry App for Windows.
---

import useBaseUrl from '@docusaurus/useBaseUrl';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<img src={useBaseUrl('img/integrations/microsoft-azure/windows.png')} alt="thumbnail icon" width="75"/> <img src={useBaseUrl('img/send-data/otel-color.svg')} alt="Thumbnail icon" width="45"/>

The Sumo Logic App for Windows allows you to monitor the performance and resource utilization of hosts and processes that your mission critical applications are dependent upon. Preconfigured dashboards provide insight into CPU, memory, network, file descriptors, page faults, and TCP connectors.

In addition to that the Windows App provides insight into your Windows system's operation and events so that you can better manage and maintain your environment. The Windows App is based on the Windows event log format and consists of predefined searches and dashboards that provide visibility into your environment for real-time analysis of overall usage of Security Status, System Activity, Updates, User Activity, and Applications.

> Windows metrics are sent to Sumo Logic through OpenTelemetry [Event Log receiver. ](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowseventlogreceiver)

Reviewer: ```suggestion
Windows event logs are sent to Sumo Logic through OpenTelemetry [Event Log receiver. ](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowseventlogreceiver)
```

Windows Host metrics are sent to Sumo Logic through OpenTelemetry [HostMetric receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver).

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Windows-OpenTelemetry/Windows-Schematics.png')} alt="Schematics" />

## Fields Created in Sumo Logic for Windows

> Following are the tags which will be created as part of Windows App install if not already present. 

Reviewer: ```suggestion
Following are the [fields](https://help.sumologic.com/docs/manage/fields/) which will be created as part of Windows App install if not already present. 
```

- **`sumo.datasource`** - Has a fixed value of **windows**.

## Log Types

The Windows App assumes events are coming from Windows Event Log receiver in JSON format. It does not work with third party logs.

Standard Windows event channels include:

-   Security
-   System
-   Application

## Collection Configuration & App installation

As part of the setting up the collection process and app installation user can select the App from App Catalog and click **Install App**. Please follow the steps below :

### Step 1: Set up Collector

> If you want to use an existing Otel Collector then this step can be skipped by selecting the option of using an existing Collector.

Reviewer: ```suggestion
:::note
If you want to use an existing OpenTelemetry Collector, you can skip this step by selecting the "Use an existing Collector" option.
:::

To create a new Collector:

1. Select the **Add a new Collector** option.
2. Select the platform for which you want to install the Sumo OpenTelemetry Collector.
```

> If you want to create a new Collector please select **Add a new Collector** option.

Reviewer: ```suggestion
```

> Select the platform for which you want to install the Sumo OpenTelemetry Collector.

Reviewer: ```suggestion
```

This will generate a command which can be executed in the machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector.  

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Windows-OpenTelemetry/Windows-Collector.png')} alt="Collector" />

### Step 2: Configure integration

In this step we will be configuring the yaml file required for Windows event logs and metrics Collection.

> Any custom tags can be added as part of this step.

Reviewer: ```suggestion
Any custom fields can be tagged along with the data in this step.
```

```

## Sample Metric Query

This sample Query is from the HostMetric - CPU > CPU User Time

### Query String

```sh
sumo.datasource=windows host.name={{host.name}} cpu=cpu0  metric=system.cpu.utilization state=user | avg by host.name
```

## Sample Logs

```sql
{"record_id":"6316","channel":"Application","event_data":"","task":"0","provider":"{\"name\":\"Microsoft-Windows-Security-SPP\",\"guid\":\"{E23B33B0-C8C9-472C-A5F9-F2BDFEA0F156}\",\"event_source\":\"Software Protection Platform Service\"}","system_time":"2023-01-20T15:22:02+0000816Z","computer":"EC2AMAZ-T30T53R","opcode":"0","keywords":"Classic","message":"Offline downlevel migration succeeded.","event_id":"{\"id\":\"16394\",\"qualifiers\":\"49152\"}","level":"Information"}
```

## Sample Log Query

This sample Query is from the Windows - Overview > System Restarts

### Query String:

```sql
%"sumo.datasource"=windows  "\"channel\":\"Security\""

> | json "event_id", "computer", "message", "channel" as event_id_obj, host, msg_summary, channel nodrop 

Reviewer: ```suggestion
| json "event_id", "computer", "message", "channel" as event_id_obj, host.name, msg_summary, channel nodrop 
```

| json field=event_id_obj "id" as event_id
| parse regex field=msg_summary "(?<msg_summary>.*\.*)" nodrop

> | where event_id = "4608" and channel = "Security" and host matches "{{host}}"

Reviewer: ```suggestion
| where event_id = "4608" and channel = "Security" and host.name matches "{{host.name}}"
```

Windows metrics are sent to Sumo Logic through OpenTelemetry [Event Log receiver. ](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/windowseventlogreceiver)

Windows Host metrics are sent to Sumo Logic through OpenTelemetry [HostMetric receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver).

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Windows-OpenTelemetry/Windows-Schematics.png')} alt="Schematics" />

## Fields Created in Sumo Logic for Windows

Following are the tags which will be created as part of Windows App install if not already present. 

- **`sumo.datasource`** - Has a fixed value of **windows**.

## Log Types

The Windows App assumes events are coming from Windows Event Log receiver in JSON format. It does not work with third party logs.

Standard Windows event channels include:

-   Security
-   System
-   Application

## Collection Configuration & App installation

> As part of the setting up the collection process and app installation user can select the App from App Catalog and click **Install App**. Please follow the steps below :

Reviewer 2: ```suggestion
As part of the setting up the collection process and app installation user can select the App from App Catalog and click **Install App**. Follow the steps below:
```
